crates/primitives/src/chain_spec.rs

```diff
@@ -0,0 +1,236 @@
+use hex_literal::hex;
+use std::collections::BTreeMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{BlockNumber, ChainId, ForkFilter, ForkHash, ForkId, Genesis, Hardfork, H256, U256};
+
+/// The Ethereum chain spec
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct ChainSpec {
+    chain_id: ChainId,
+    genesis: Genesis,
+    genesis_hash: H256,
+    hardforks: BTreeMap<Hardfork, BlockNumber>,
+    // This is an edge case, maybe we move it into a `Hardforks` struct
+    paris_block: Option<u64>,
+    paris_ttd: Option<U256>,
+    shanghai_block: Option<u64>,
+}
+
+impl ChainSpec {
+    /// Returns the Etereum mainnet spec
+    pub fn mainnet() -> Self {
```

Comment:
Would move this out to a `const` if possible:

```rs
pub const MAINNET: ChainSpec = ChainSpec {
  // ...
};
```

Reply:
I wasn't able to move `mainnet()` to a const because `BTreeMap::from()` isn't a const fn (same for `Genesis::default()` but it's temporary).

Maybe it can be done with something like `lazy_static`...

Reply:
Ah, right. I think a lazy static is OK.